src/fixed_decimals_utils.jl

```diff
@@ -1,410 +1,508 @@
             else
-                #parse error leading zero 00.x...
-                return (T(0), code | Parsers.INVALID, field_end)
+                # We're done with parsing the integer part, now we need to check if we have a decimal point,
+                # exponent, or it there is something else and should yield back to Parsers.jl to recover.
+                break
             end
-        else
-            return (T(0), code | Parsers.OK, field_end) # ok
+            prev_b = b
         end
     end
+    last_byte_to_parse = found_nonzero * (pos - 1)
 
-    number_of_fractional_digits -= exp_val
-    number_of_digits = last_byte_to_parse - pos + 1 - parse_through_decimal
-    if has_digits_past_decimal
-        number_of_fractional_digits += last_byte_to_parse - decimal_position
-    end
-
-    decimal_shift = f - number_of_fractional_digits
-
-    backing_integer_digits = number_of_digits - number_of_fractional_digits + f - ngroupmarks
-    digits_to_iter = min(backing_integer_digits, number_of_digits)
+    # The fractional digits
+    if b == options.decimal
+        prev_b == groupmark && @goto invalid # Input is something like "10,."
+        decimal_position = pos
+        pos += 1
+        Parsers.incr!(source)
+        Parsers.eof(source, pos, len) && (code |= Parsers.EOF; @goto done_scanning) # Input is something like "1.<EOF>"
+        b = Parsers.peekbyte(source, pos)
 
-    if backing_integer_digits < 0 # All digits are past our precision, no overflow possible
-        return (T(0), code | Parsers.OK, field_end)
-    elseif backing_integer_digits == 0 # All digits are past our precision but we may get a 1 from rounding, no overflow possible
-        out = T(0)
-        if number_of_fractional_digits != f
-            round_val, code = _parse_round(T, buf, pos + parse_through_decimal, last_byte_to_parse, decimal_position, code, mode)
-            out += int_sign * round_val
-        end
-    elseif backing_integer_digits < maxdigits(T) # The number of digits to accumulate is smaller than the capacity of T, no overflow possible
-        out = zero(T)
-        for i in (pos:pos - 1 + digits_to_iter + parse_through_decimal + ngroupmarks)
-            (i == decimal_position) && continue
-            b = buf[i]
-            b == groupmark && continue
-            out = T(10) * out + T(b - UInt8('0'))
-        end
-        out *= int_sign
-        if number_of_fractional_digits < f
-            out = _shift(out, decimal_shift)
-        elseif number_of_fractional_digits != f
-            round_val, code = _parse_round(T, buf, pos + digits_to_iter + parse_through_decimal + ngroupmarks, last_byte_to_parse, decimal_position, code, mode)
-            out += int_sign * round_val
+        # If we started on a decimal point or with leading zeros only, we'll keep skipping zeros
+        if !found_nonzero
+            found_digit |= b == UInt8('0')
```

Comment:
This might be easier to read with parens:
```suggestion
            found_digit |= (b == UInt8('0'))
```